Create key ID index to find user names / pubkeys

BobWall23 opened the pull request:
Any time we fetch a pubkey, generate an index entry that maps the key ID to the user@host name of the pubkey.

Added commands finduser (to fetch pubkey without adding recipient), clrrcpt (to clear recipients), showrcpt (to show current list of recipients).

When downloading an encrypted file, lists the logins of the users where key IDs can be resolved using the index. Lists the count of key IDs that could not be resolved.

BobWall23 committed:
Create key ID index to find user names / pubkeys

Any time we fetch a pubkey, generate an index entry that maps the
key ID to the user@host name of the pubkey.
Added commands finduser (to fetch pubkey without adding recipient),
clrrcpt (to clear recipients), showrcpt (to show current list of
recipients).
When downloading an encrypted file, lists the logins of the users where
key IDs can be resolved using the index. Lists the count of key IDs that
could not be resolved.

BobWall23 committed:
detab

zmre commented on iron/recipient.c:
```diff
@@ -209,8 +213,28 @@ iron_remove_recipient(const char * login)
+ *  For each of the entries (including the current user), display the login.
+ */
+void
+iron_list_recipients(void)
```
Shouldn't this function filter by the current host?

BobWall23 replied:
The recipients are state for the current session, and the current session is only against one host.

zmre commented on sftp.c:
```diff
@@ -196,9 +199,15 @@ static const struct CMD cmds[] = {
 	{ "dir",	I_LS,		REMOTE	},
 	{ "exit",	I_QUIT,		NOARGS	},
+#ifdef IRONCORE
+	{ "finduser",	I_FIND_USER,	LOCAL	},
```
Why is this marked as `LOCAL`? Isn't it looking at the remote filesystem?

BobWall23 replied:
Yeah, that sounds right.

zmre commented on sftp.c:
```diff
@@ -1778,6 +1797,20 @@ parse_dispatch_command(struct sftp_conn *conn, const char *cmd, char **pwd,
+		iron_reset_recipients();
+		break;
+	case I_SHOW_RCPT:
+		iron_list_recipients();
```
Naming nitpick: I find it weird that the command is `showrcpt` and the function is `iron_list_recipients`. Feels like the function should be `iron_show_recipients` or else the command should be `lsrcpt` or something.

BobWall23 replied:
The command isn't `lsrcpt` because all commands in sftp that start with `l` are "local" commands (e.g. `lpwd`, `lis`, `lcd`).
I'll change the function names.

BobWall23 committed:
Code review changes.

Made I_FINDUSR REMOTE.
Changed iron_list_recipients to iron_show_recipients
Changed iron_reset_recipients to iron_clear_recipients
